runtime/lua/vim/iter.lua

```diff
@@ -863,6 +863,48 @@ function ListIter.enumerate(self)
   return self
 end
 
+--- Each element of iter is concatenated and returned.
+--- sep, i, j are the same as the arguments to |table.concat()|.
+---
+--- The following two are equivalent
+---
+--- Use table.concat()
+--- <pre>
+--- table.concat(vim
+---   .iter({ 1, 2, 3 })
+---   :map(function(v)
+---     return v * 2
+---   end)
+---   :totable(), ', ')
+--- </pre>
+---
+--- Use Iter:join()
+--- <pre>
+--- vim
+---   .iter({ 1, 2, 3 })
+---   :map(function(v)
+---     return v * 2
+---   end)
+---   :join(', ')
+--- </pre>
+---@param sep string|nil
+---@param i integer|nil
+---@param j integer|nil
+---@return string
+function Iter.join(self, sep, i, j)
+  local t = self
+    :map(function(_, v)
+      return v
+    end)
+    :totable()
```

Comment:
I think `totable()` could be avoided, no? Concatenation can be "streamed". We should always consider that when enhancing `vim.iter`.